Gem/Code/Source/Components/PerfTest/NetworkPrefabSpawnerComponent.cpp

```diff
@@ -0,0 +1,245 @@
+
+        if (newAsset.m_spawnableAsset.IsReady() == false)
+        {
+            newAsset.m_spawnableAsset.QueueLoad();
+        }
+
+        m_assetMap.emplace(newAsset.m_spawnableAsset.GetId(), newAsset);
+
+        const SpawnRequest request{ newAsset.m_spawnableAsset.GetId(), worldTm, AZStd::move(callbacks) };
+        if (newAsset.m_spawnableAsset.IsReady())
+        {
+            CreateInstance(request, &newAsset);
+        }
+        else
+        {
+            m_requests.push_back(request);
+        }
+    }
+
+    AZ::Data::AssetId NetworkPrefabSpawnerComponent::GetSpawnableAssetId(const char* assetPath) const
+    {
+        if (assetPath)
+        {
+            AZ::Data::AssetId assetId;
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                assetId, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetIdByPath, assetPath,
+                AZ::Data::s_invalidAssetType, false);
+            if (assetId.IsValid())
+            {
+                return assetId;
+            }
+        }
+
+        return {};
+    }
+
+    void NetworkPrefabSpawnerComponent::CreateInstance(const SpawnRequest& request, const AssetItem* asset)
+    {
+        AZ::Transform world = request.m_whereToSpawn;
+
+        if (asset)
+        {
+            auto ticket = AZStd::make_shared<AzFramework::EntitySpawnTicket>(asset->m_spawnableAsset);
+
+            auto preSpawnCallback = [world, request, ticket]([[maybe_unused]] AzFramework::EntitySpawnTicket::Id ticketId, AzFramework::SpawnableEntityContainerView view)
+            {
+                const AZ::Entity* rootEntity = *view.begin();
+                if (AzFramework::TransformComponent* entityTransform = rootEntity->FindComponent<AzFramework::TransformComponent>())
+                {
+                    entityTransform->SetWorldTM(world);
+                }
+
+                if (request.m_callbacks.m_beforeActivateCallback)
+                {
+                    request.m_callbacks.m_beforeActivateCallback(ticket, view);
+                }
+            };
+
+            auto onSpawnedCallback = [request, ticket]([[maybe_unused]] AzFramework::EntitySpawnTicket::Id ticketId, AzFramework::SpawnableConstEntityContainerView view)
+            {
+                if (request.m_callbacks.m_onActivateCallback)
+                {
+                    request.m_callbacks.m_onActivateCallback(ticket, view);
+                }
+            };
+
+            if (ticket->IsValid())
+            {
+                AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+                optionalArgs.m_preInsertionCallback = AZStd::move(preSpawnCallback);
+                optionalArgs.m_completionCallback = AZStd::move(onSpawnedCallback);
+                AzFramework::SpawnableEntitiesInterface::Get()->SpawnAllEntities(*ticket, AZStd::move(optionalArgs));
+            }
+            else
+            {
+                AZ_Assert(ticket->IsValid(), "Unable to instantiate spawnable asset");
```

Comment:
Could place the assert above the if or always assert False if you’re already in the else

Reply:
Done.